platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -0,0 +1,120 @@
+#import "MGLMapSnapshotter.h"
+
+#import <mbgl/actor/actor.hpp>
+#import <mbgl/actor/scheduler.hpp>
+#import <mbgl/util/geo.hpp>
+#import <mbgl/map/map_snapshotter.hpp>
+#import <mbgl/map/camera.hpp>
+#import <mbgl/storage/default_file_source.hpp>
+#import <mbgl/util/default_thread_pool.hpp>
+#import <mbgl/util/string.hpp>
+#import <mbgl/util/shared_thread_pool.hpp>
+
+#import "MGLOfflineStorage_Private.h"
+#import "MGLGeometry_Private.h"
+#import "UIImage+MGLAdditions.h"
+
+@implementation MGLMapSnapshotOptions
+
+- (instancetype _Nonnull)initWithStyleURL:(NSURL* _Nonnull) styleURL mapCamera:(MGLMapCamera*) mapCamera size:(CGSize) size;
+{
+    self = [super init];
+    if (self) {
+        _styleURL = styleURL;
+        _size = size;
+        _mapCamera = mapCamera;
+        _scale = [UIScreen mainScreen].scale;
+    }
+    return self;
+}
+
+@end
+
+@implementation MGLMapSnapshotter {
+    
+    std::shared_ptr<mbgl::ThreadPool> mbglThreadPool;
+    std::unique_ptr<mbgl::MapSnapshotter> mbglMapSnapshotter;
+    std::unique_ptr<mbgl::Actor<mbgl::MapSnapshotter::Callback>> snapshotCallback;
+    
+    BOOL loading;
```

Comment:
Nit: loading is already synthesized. This can be dropped. Use `_loading` to ignore the accessor when you set the variable.